engine/classes/Elgg/Router.php

```diff
@@ -260,9 +289,13 @@ public function registerPageHandler($identifier, $function) {
 	 */
 	public function registerRoute($name, array $params = []) {
 
+		$params = $this->hooks->trigger('route:config', $name, $params, $params);
+
 		$path = elgg_extract('path', $params);
 		$resource = elgg_extract('resource', $params);
 		$handler = elgg_extract('handler', $params);
+		$middleware = elgg_extract('middleware', $params, []);
+		$public = elgg_extract('public', $params, false);
```

Comment:
why default to non-public routes, doesn't that cause lots of problems

Reply:
That's the current approach. When you enable WG everything is non-public, unless allowed by hook

Reply:
ah so this only gets enforced by WG (and added middleware)?

Reply:
I suppose "public" is ambiguous. Should rename into something else. Ideas?